### PR TITLE
Signed int overflow lib changes and tests

### PR DESCRIPTION
**Pull request by timmui**
* Fixes from #112 and unit tests for the signed integer overflow library
* Clarified library and changed return values (breaking change)
* Added unit tests

### CODE REVIEW

**Comment by timmui** [comment]
@markhannum

File: bbinc/int_overflow.h [hunk 1]
```diff
@@ -20,10 +20,10 @@
 extern "C" {
```
Comment (nirbhayc) [inline bbinc/int_overflow.h]: I am wondering if this is really needed.

**Comment by nirbhayc** [comment]
Looks good.

File: bbinc/int_overflow.h (hunk 1 above)
Comment (timmui) [inline bbinc/int_overflow.h]: Not sure. I've seen other files in `/bbinc` with it so I did the same

**Comment by markhannum** [comment]
agree - looks great

**Pull request merged**